### PR TITLE
Book detail screen

### PR DESCRIPTION
**Pull request by felire**
### Summary

I made the DetailBox Component. This component show the image and some info about the book. Also, show the 'Add to wishlist' button and the 'rent' button. The rent button change if the book is available or not available. 

### Gifs

Not Available book

transition2

Available book

transition3

### CODE REVIEW

File: WBooks/src/app/screens/BookDetail/components/DetailBox/styles.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { StyleSheet } from 'react-native';
+
+import { white, darkGray, red, green, blue, lightGray, veryBlue } from '../../../../../constants/colors';
+
+const styles = StyleSheet.create({
+  bookContainer: {
+    flexDirection: 'column',
+    paddingTop: 15,
```
Comment (santinucera) [inline WBooks/src/app/screens/BookDetail/components/DetailBox/styles.js]: Check if you can have all paddings in one line. Check padding documentation

File: WBooks/src/app/screens/BookDetail/components/DetailBox/styles.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    fontSize: 15
+  },
+  type: {
+    color: darkGray,
+    fontSize: 15
+  },
+  buttonsContainer: {
+    marginTop: 15
+  },
+  genericStyleButton: {
```
Comment (santinucera) [inline WBooks/src/app/screens/BookDetail/components/DetailBox/styles.js]: Same comment of above for padding

File: WBooks/src/app/screens/BookDetail/components/DetailBox/index.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  render() {
+    const { navigation } = this.props;
+    const book = navigation.getParam('book', {});
+    const { image_url: image, title, author, year } = book;
+    const availabilityStyle = genericBook.isAvailable ? styles.available : styles.notAvailable;
+    const availabilityText = genericBook.isAvailable ? availability.available : availability.notAvailable;
+    const rentButtonStyle = genericBook.isAvailable
+      ? styles.rentButtonAvailable
+      : styles.rentButtonNotAvailable;
+    const textButtonRentStyle = genericBook.isAvailable ? styles.textAvailable : styles.textNotAvailable;
```
[the remark this answers is not in the document]
Comment (felire) [reply WBooks/src/app/screens/BookDetail/components/DetailBox/index.js]: The only idea to resolve this comment is abstract all that login in one if/else block. Do you have any other idea?

File: WBooks/src/app/screens/BookDetail/components/DetailBox/styles.js (hunk 1 above)
Comment (felire) [reply WBooks/src/app/screens/BookDetail/components/DetailBox/styles.js]: I tried to do that but I can't find something as simple as the css way. I hope to find something

Comment (felire) [reply WBooks/src/app/screens/BookDetail/components/DetailBox/styles.js]: It exists paddingVertical and paddingHorizontal, I use them?

File: WBooks/src/app/screens/BookDetail/components/DetailBox/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (santinucera) [reply WBooks/src/app/screens/BookDetail/components/DetailBox/index.js]: I have in mind, to abstract this logic in a function that only do that if in line.

File: WBooks/src/app/screens/BookDetail/components/DetailBox/styles.js (hunk 1 above)
Comment (santinucera) [reply WBooks/src/app/screens/BookDetail/components/DetailBox/styles.js]: I think it will be better

**Pull request merged**